pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java

```diff
@@ -387,15 +391,15 @@ private SegmentMetadata getSegmentMetadata(File tempDecryptedFile, File tempSegm
     return MetadataExtractorFactory.create(metadataProviderClass).extractMetadata(tempDecryptedFile, tempSegmentDir);
   }
 
-  private void completeZkOperations(boolean enableParallelPushProtection, HttpHeaders headers, File uploadedSegmentFile,
-      String tableNameWithType, SegmentMetadata segmentMetadata, String segmentName, String zkDownloadURI,
-      boolean moveSegmentToFinalLocation, String crypter)
+  private boolean completeZkOperations(boolean enableParallelPushProtection, HttpHeaders headers,
```

Comment:
If you want to change the function to return `boolean`, I think that we should add the documentation on when we return `true` and on what situations we would return `false.`

Reply:
Added comment.